src/main/java/com/hotels/hcommon/ssh/SshSettings.java

```diff
@@ -90,18 +101,25 @@ public static Builder builder() {
 
   private final int sshPort;
   private final String route;
-  private final List<String> privateKeys;
+  private List<String> privateKeys;
   private final String knownHosts;
+  private final String localhost;
   private final int sessionTimeout;
   private final boolean strictHostKeyChecking;
 
-  private SshSettings(Builder builder) {
+  protected SshSettings(Builder builder) {
     sshPort = builder.sshPort;
     route = builder.route;
-    privateKeys = Collections.unmodifiableList(Arrays.asList(builder.privateKeys.split(",")));
     knownHosts = builder.knownHosts;
+    localhost = builder.localHost;
     sessionTimeout = builder.sessionTimeout;
     strictHostKeyChecking = builder.strictHostKeyChecking;
+
+    try {
+      privateKeys = Collections.unmodifiableList(Arrays.asList(builder.privateKeys.split(",")));
```

Comment:
Is there something we can check with == null in an if/else rather than try/catch on an NPE? Also, do you have a test for this case?